### PR TITLE
Refactor task create

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/workflow/src/records/fhir.ts [hunk 1]
```diff
@@ -1173,7 +1167,7 @@ export function createCorrectionRequestTask(
 function createNewTaskResource(
   previousTask: SavedTask,
   newExtensions: Extension[],
-  practitionerId: Practitioner['id'],
+  practitionerId?: Practitioner['id'],
```
Comment (Zangetsu101) [inline packages/workflow/src/records/fhir.ts:1170]: Is there a need to even keep this argument? Like we only seem to be passing "undefined" as the parameter

File: packages/workflow/test/handlers.ts [hunk 2]
```diff
@@ -116,9 +117,102 @@ const duplicatesHandler = rest.post(
...
+      },
+      {
+        response: {
+          status: '201',
+          location:
+            '/fhir/Location/ed6195ff-0f83-4852-832e-dc9db07151ff/_history/7d23bdbd-d987-4c03-8cad-1d4e4b7b986c' as URLReference
+        }
+      }
+    ],
+    type: 'transaction-response'
```
Comment (Zangetsu101) [inline packages/workflow/test/handlers.ts:213]: Not sure if it makes sense to have the full bundle response by default here

File: packages/workflow/src/records/fhir.ts (hunk 1 above)
Comment (Nil20) [reply packages/workflow/src/records/fhir.ts:1170]: Yeah for download, unassign, view endpoint's record, where we don't merge the inputBundle with heath response.

File: packages/workflow/test/handlers.ts (hunk 2 above)
Comment (Nil20) [reply packages/workflow/test/handlers.ts:213]: Yeah! Correct..

File: packages/workflow/src/records/fhir.ts (hunk 1 above)
Comment (Nil20) [reply packages/workflow/src/records/fhir.ts:1170]: Or we could use the `withPractitionerDetails` for them as well and remove the practitionerId param from here.

File: packages/workflow/src/records/fhir.ts [hunk 3]
```diff
@@ -881,9 +785,23 @@ export function createRegisterTask(
   }
 }
 
+export function createReinstateTask(
+  previousTask: SavedTask,
+  prevBusinessStatuses: RegistrationStatus
```
Comment (Zangetsu101) [inline packages/workflow/src/records/fhir.ts:790]:
```suggestion
  prevBusinessStatus: RegistrationStatus
```

File: packages/workflow/src/records/handler/reinstate.ts [hunk 4]
```diff
@@ -63,13 +57,6 @@ export const reinstateRoute = createRoute({
       throw new Error('No task found after reinstating')
     }
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/reinstate.ts:58]: We probably don't need this

File: packages/workflow/src/records/handler/unassign.ts [hunk 5]
```diff
@@ -45,10 +44,7 @@ export async function unassignRecordHandler(
     ]
   )
 
-  const unassignedRecordWithTaskOnly = await toUnassigned(
-    record,
-    await getLoggedInPractitionerResource(token)
-  )
+  const unassignedRecordWithTaskOnly = await toUnassigned(record, token)
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/unassign.ts:47]:
```suggestion
  const unassignedRecord = await toUnassigned(record, token)
```

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**